x-pack/plugins/ingest_manager/common/services/is_valid_namespace.ts

```diff
@@ -3,15 +3,44 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+import { i18n } from '@kbn/i18n';
 
 // Namespace string eventually becomes part of an index name. This method partially implements index name rules from
 // https://github.com/elastic/elasticsearch/blob/master/docs/reference/indices/create-index.asciidoc
-export function isValidNamespace(namespace: string) {
-  return (
-    typeof namespace === 'string' &&
-    // Lowercase only
-    namespace === namespace.toLowerCase() &&
-    // Cannot include \, /, *, ?, ", <, >, |, space character, comma, #, :
-    /^[^\*\\/\?"<>|\s,#:]+$/.test(namespace)
-  );
+// and implements a limit based on https://github.com/elastic/kibana/issues/75846
+export function isValidNamespace(namespace: string): { valid: boolean; error?: string } {
+  if (!namespace.trim()) {
+    return {
+      valid: false,
+      error: i18n.translate('xpack.ingestManager.namespaceValidation.requiredErrorMessage', {
+        defaultMessage: 'Namespace is required',
+      }),
+    };
+  } else if (namespace !== namespace.toLowerCase()) {
+    return {
+      valid: false,
+      error: i18n.translate('xpack.ingestManager.namespaceValidation.lowercaseErrorMessage', {
+        defaultMessage: 'Namespace must be lowercase',
+      }),
+    };
+  } else if (/[\*\\/\?"<>|\s,#:]+/.test(namespace)) {
+    return {
+      valid: false,
+      error: i18n.translate(
+        'xpack.ingestManager.namespaceValidation.invalidCharactersErrorMessage',
+        {
+          defaultMessage: 'Namespace contains invalid characters',
+        }
+      ),
+    };
+  } else if (new Blob([namespace]).size > 100) {
+    return {
+      valid: false,
+      error: i18n.translate('xpack.ingestManager.namespaceValidation.tooLongErrorMessage', {
+        defaultMessage: 'Namespace is too long',
```

Comment:
Could you add here that it has a max of 100 chars?

Reply:
100 characters is not totally accurate as it tests on bytes, but displaying a message about bytes may not be very user-friendly.

@dedemorton / @gchaps any suggestions for an error message here? this is a text field that cannot be more than 100 bytes long (non-ASCII characters are multibyte)

Reply:
How about one of the following:

Namespace cannot be more than 100 bytes.

Namespace cannot be more than 100 bytes.  Non-ASCII characters are multibyte.

Namespace cannot be more than 100 bytes.  Non-ASCII characters might require 2 to 4 bytes.  (or what the number might be).